arborist/server_test.go

```diff
@@ -3074,91 +3073,177 @@ func TestServer(t *testing.T) {
 				grantUserPolicy(t, username, policyName, "null")
 			})
 
-			t.Run("POST_usernameProvided", func(t *testing.T) {
-				w := httptest.NewRecorder()
-				body := []byte(fmt.Sprintf(`{"username": "%s"}`, username))
-				req := newRequest("POST", "/auth/mapping", bytes.NewBuffer(body))
-				handler.ServeHTTP(w, req)
-				// expect to also receive auth mappings of anonymous and logged-in policies
-				testAuthMappingResponse(t, w)
-			})
+			t.Run("POST_body", func(t *testing.T) {
+				t.Run("usernameProvided", func(t *testing.T) {
+					w := httptest.NewRecorder()
+					body := []byte(fmt.Sprintf(`{"username": "%s"}`, username))
+					req := newRequest("POST", "/auth/mapping", bytes.NewBuffer(body))
+					handler.ServeHTTP(w, req)
+					// expect to also receive auth mappings of anonymous and logged-in policies
+					testUserAuthMappingResponse(t, w)
+				})
 
-			t.Run("POST_userDoesNotExist", func(t *testing.T) {
-				w := httptest.NewRecorder()
-				badUsername := "hulkhogan12"
-				body := []byte(fmt.Sprintf(`{"username": "%s"}`, badUsername))
-				req := newRequest("POST", "/auth/mapping", bytes.NewBuffer(body))
-				handler.ServeHTTP(w, req)
+				t.Run("userDoesNotExist", func(t *testing.T) {
+					w := httptest.NewRecorder()
+					badUsername := "hulkhogan12"
+					body := []byte(fmt.Sprintf(`{"username": "%s"}`, badUsername))
+					req := newRequest("POST", "/auth/mapping", bytes.NewBuffer(body))
+					handler.ServeHTTP(w, req)
+					assert.Equal(t, w.Code, http.StatusOK, "expected a 200 OK")
 
-				// expect a 200 OK response
-				assert.Equal(t, w.Code, http.StatusOK, "expected a 200 OK")
+					// expect result to only contain anonymous and loggedIn auth mappings.
+					result := make(arborist.AuthMapping)
+					err = json.Unmarshal(w.Body.Bytes(), &result)
+					if err != nil {
+						httpError(t, w, "couldn't read response from auth mapping")
+					}
+					expectedMappings := make(arborist.AuthMapping)
+					for k, v := range anonymousAuthMapping {
+						expectedMappings[k] = v
+					}
+					for k, v := range loggedInAuthMapping {
+						expectedMappings[k] = v
+					}
+					msg := fmt.Sprintf("Expected response to be these auth mappings from anonymous and logged-in groups: %v", expectedMappings)
+					for resource, actions := range result {
+						assert.Contains(t, expectedMappings, resource, msg)
```

Comment:
Just for my understanding - if this "hulkhogan12" has auth mappings other than loggedIn or anonymous, this assert.Contains will fail. Is that correct?

Reply:
yes 👍